pom.xml

```diff
@@ -0,0 +1,233 @@
+<!--                        <version>1.5</version> &lt;!&ndash; must match java/pom.xml &ndash;&gt;-->
+<!--                        <configuration>-->
+<!--                            <keyname>C2F054455909C25FE818EE9B93488CAD47AEC27B</keyname>-->
+<!--                        </configuration>-->
+<!--                        <executions>-->
+<!--                            <execution>-->
+<!--                                <id>sign-artifacts</id>-->
+<!--                                <phase>verify</phase>-->
+<!--                                <goals>-->
+<!--                                    <goal>sign</goal>-->
+<!--                                </goals>-->
+<!--                            </execution>-->
+<!--                        </executions>-->
+<!--                    </plugin>-->
+<!--                </plugins>-->
+<!--            </build>-->
+<!--        </profile>-->
+    </profiles>
+
+    <distributionManagement>
+        <snapshotRepository>
+            <id>ossrh</id>
+            <name>Nexus Snapshots Repo</name>
+            <url>https://s01.oss.sonatype.org/content/repositories/snapshots/</url>
+        </snapshotRepository>
+        <repository>
+            <id>ossrh</id>
+            <name>Nexus Staging Repo</name>
+            <url>https://s01.oss.sonatype.org/nexus/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.12.4</version>
+<!--                <configuration>-->
+<!--                    <excludes>-->
+<!--                        <exclude>${exclude.tests}</exclude>-->
+<!--                    </excludes>-->
+<!--                </configuration>-->
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.sonatype.plugins</groupId>
+                <artifactId>nexus-staging-maven-plugin</artifactId>
+                <version>1.6.8</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <serverId>ossrh</serverId>
+                    <nexusUrl>https://s01.oss.sonatype.org/</nexusUrl>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <version>2.7.7</version> <!-- The common Hadoop version for Spark 2.4.7 and 3.0.1, which lakeFS supports -->
```

Comment:
Hopefully we don't tie ourselves to hardcoded Hadoop versions this time.
For now we can define a variable `hadoop.version`.

Also, we need an action item to discover whether or not this dependency should be _provided_.

Reply:
> Hopefully we don't tie ourselves to hardcoded Hadoop versions this time.
For now we can define a variable hadoop.version.

sounds good, let me know if the change I made is what you expected to see. for now, i'm using hadoop 2.7.7 as the variable value. 

> Also, we need an action item to discover whether or not this dependency should be provided.

The code does not compile without providing this dependency. we need it to import the hadoopfs class we override. 
Let me know if you still think that we need a discovery.